lib/pipeline/webpack.config.js

```diff
@@ -0,0 +1,187 @@
+/* global module process require */
+
+// NOTE: some packages, plugins, and presets referenced/required in this webpack
+// config are deps of embark and will be transitive dapp deps unless specified
+// in the dapp's own package.json, perhaps with a different version/range or
+// other specifier
+
+// NOTE: embark modifies process.env.NODE_PATH so that when running dapp
+// scripts in embark's child processes, embark's own node_modules directory
+// will be searched by node's require(); that allows an ejected version of this
+// config, inside a dapp, to function correctly without embark being an
+// explicit dependency of the dapp; however, webpack and babel do not directly
+// support NODE_PATH, so in various parts of the config, modules such as babel
+// plugins and presets must be resolved with require.resolve(); that is only
+// necessary if a plugin/preset is in embark's node_modules vs. the dapp's
+// node_modules
+
+// there's a bug in pkg clone-deep re: regex; for now use lodash.clonedeep
+// see: https://github.com/jonschlinkert/clone-deep/pull/14
+const cloneDeep = require('lodash.clonedeep');
+const CompressionPlugin = require('compression-webpack-plugin');
+const glob = require('glob');
+const HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
+const path = require('path');
+
+const dappPath = process.env.DAPP_PATH;
+const embarkPath = process.env.EMBARK_PATH;
+
+const embarkAliases = require(path.join(dappPath, '.embark/embark-aliases.json'));
+const embarkAssets = require(path.join(dappPath, '.embark/embark-assets.json'));
+const embarkNodeModules = path.join(embarkPath, 'node_modules');
+const embarkJson = require(path.join(dappPath, 'embark.json'));
+
+const buildDir = path.join(dappPath, embarkJson.buildDir);
+
+// it's important to `embark reset` if a pkg version is specified in
+// embark.json and changed/removed later, otherwise pkg resolution may behave
+// unexpectedly, e.g. a different version may get used owing to lexographical
+// order or because .embark/versions is still in place even though defaults are
+// desired; once embark and its dependencies are always resolvable (in the
+// usual node sense) relative to a dapp, embark's pkg version-management
+// functionality can probably be deprecated in favor of a dapp simply
+// specifying dependencies in its own package.json, e.g. a different ipfs
+// version than embark's default
+let versions;
+try {
+  versions = glob.sync(path.join(dappPath, '.embark/versions/*/*'));
+} catch (e) {
+  versions = [];
+}
+
+const entry = Object.keys(embarkAssets)
+      .filter(key => key.match(/\.js?$/))
```

Comment:
is this normal indentation?

Reply:
It's what `js2-mode` insisted upon, though I could make it two-spaces. Preference?

Reply:
Yes 2 spaces is how I would expect it to be:
```
const entry = Object.keys(embarkAssets)
  .filter(key => key.match(/\.js?$/))
  .reduce((obj, key) => {
    obj[key] = embarkAssets[key]
      .map(file => {
        let file_path = file.path;
        if (!file.path.match(/^\.\//)) {
          file_path = './' + file_path;
        }
        return file_path;
      });
    return obj;
  }, {});
```

Reply:
Done.